effects/data_base_effect.py

```diff
@@ -0,0 +1,71 @@
+from typing import List
+
+import message_text_config as config
+import user_interaction
+import utils
+from effects.effect import Effect
+from message_text_config import Errors
+
+
+class DataBase(Effect):
+    def __init__(self, game: Game, name: str, creator: Citizen) -> None:
+        super(DataBase, self).__init__(game, name, creator, 4)
+
+    def _activate_impl(self):
+        target_numbers = []
+        for _ in range(3):
+            target_number = user_interaction.read_index("config.type_target")
+            target_numbers.append(target_number)
+
+        while (not self._validate(target_numbers)):
+            target_number = user_interaction.read_index("config.type_target")
+            target_numbers = []
+            for _ in range(3):
+                target_number = user_interaction.read_index(
+                    "config.type_target")
+                target_numbers.append(target_number)
+
+        self._targets = [
+            self._game.citizens[target_number - 1]
+            for target_number in target_numbers
+        ]
+        return True
+
+    def _validate(self, target_numbers: List[int]):
+        # TODO utils methods shood show message errors
+
+        is_in_range = [
+            all(
+                utils.validate_citizen_target_number(target,
+                                                     self._game.citizens))
+            for target in target_numbers
+        ]
+        if not is_number:
+            return False
+
+        is_three_targets = (len(target_numbers) == 3)
+        if not is_three_targets:
+            user_interaction.show_active_instant(Errors.DATA_BASE_TARGETS)
+            return False
+
+        is_different = (len(target_numbers) == len(set(target_numbers)))
+        if not is_different:
+            user_interaction.show_active_instant("Цели не должны совпадать")
+            return False
+
+        return True
+
+    def _resolve_impl(self):
+        roles = [type(target).__name__ for target in self._targets]
+        if ("Player" in roles) and ("Spy" in roles):
```

Comment:
I'd prefere actual types, not.type names. Would be easier to refactor if some of the names will change. But it is also ok.